presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -181,46 +179,92 @@ public static BytecodeNode generateInvocation(Scope scope, String name, ScalarFu
         // Index of parameter (without @IsNull) in Presto function
         int realParameterIndex = 0;
 
+        // go through all the choices in the function and then pick the best one
+        List<ScalarImplementationChoice> choices = function.getAllChoices();
+        int bestChoiceIndex = -1;
+        for (int i = 0; i < choices.size(); i++) {
+            boolean isInvalid = false;
+            ScalarImplementationChoice current = choices.get(i);
+            for (int j = 0; j < arguments.size(); j++) {
+                boolean isInputReference = false;
+                boolean isValueType = current.getArgumentProperty(j).getArgumentType() == VALUE_TYPE;
+                if (!isValueType) {
+                    continue;
+                }
+                if (arguments.get(j) instanceof InputReferenceNode) {
+                    isInputReference = true;
+                }
+                NullConvention currentNullConvention = current.getArgumentProperty(j).getNullConvention();
+                if (!isInputReference && currentNullConvention == NullConvention.BLOCK_AND_POSITION) {
+                    isInvalid = true;
+                    break;
+                }
+            }
+            if (!isInvalid) {
+                bestChoiceIndex = i;
+            }
+        }
+
+        checkArgument(bestChoiceIndex >= 0, "None of the choices are valid");
+        ScalarImplementationChoice bestChoice = choices.get(bestChoiceIndex);
```

Comment:
`bestChoiceIndex` seems unnecessary. Why not update `bestChoice` in the above loop directly instead?